File: Src/Icm.Core/Localization/ILocalizationServiceExtensions.vb
Imports System.Runtime.CompilerServices

Namespace Icm.Localization
    Public Module ILocalizationServiceExtensions

        <Extension>
        Public Function TransAnd(locService As ILocalizationService, ParamArray args() As Object) As String
            Return New PhraseAnd(args).Translate(locService.Lcid, locService.Repository)
        End Function

        <Extension>
        Public Function TransF(locService As ILocalizationService, key As String, ParamArray args() As Object) As String
            Return New PhraseFormat(key, args).Translate(locService.Lcid, locService.Repository)
        End Function

        <Extension>
        Public Function Trans(locService As ILocalizationService, key As String) As String
            Return locService.Repository.ItemForCulture(locService.Lcid, key)
        End Function
    End Module
End Namespace

File: Src/Icm.Core/Localization/ILocalizationServiceExtensions.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Runtime.CompilerServices;

namespace Icm.Localization
{
	public static class ILocalizationServiceExtensions
	{

		[Extension()]
		public static string TransAnd(ILocalizationService locService, params object[] args)
		{
			return new PhraseAnd(args).Translate(locService.Lcid, locService.Repository);
		}

		[Extension()]
		public static string TransF(ILocalizationService locService, string key, params object[] args)
		{
			return new PhraseFormat(key, args).Translate(locService.Lcid, locService.Repository);
		}

		[Extension()]
		public static string Trans(ILocalizationService locService, string key)
		{
			return locService.Repository.ItemForCulture(locService.Lcid, key);
		}
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
